csharp/private/tools/assembly-runner/lib.bzl

```diff
@@ -0,0 +1,147 @@
+load(
+    "@d2l_rules_csharp//csharp/private:providers.bzl",
+    "AnyTargetFramework",
+    "CSharpAssembly",
+)
+
+def _write_wrapper_cc_impl(ctx):
+    provider = CSharpAssembly[ctx.attr.target_framework]
+    exe = ctx.attr.exe[provider]
+
+    toolchain = ctx.toolchains["@d2l_rules_csharp//csharp/private:toolchain_type"]
+
+    ctx.actions.expand_template(
+        template = ctx.files._main_cc[0],
+        output = ctx.outputs.out,
+        substitutions = {
+            "{TargetExe}": exe.out.short_path,
+
+            # Strip the leading "../" off the path to dotnet.exe
+            "{DotnetExe}": toolchain.runtime.short_path[3:],
```

Comment:
I think using the toolchain like this might be OK... I'm worried about a situation where our toolchain doesn't match the uh... OS we run `bazel run` on (remote builds?) and we end up picking the wrong one.
We use the toolchain again in `_csharp_exe` with the new `everything` filegroup target.